tests/suites/spaces_ec2/juju_bind.sh

```diff
@@ -0,0 +1,57 @@
+run_juju_bind_ec2() {
+    echo
+
+    file="${TEST_DIR}/test-juju-bind-ec2.txt"
+
+    ensure "spaces-juju-bind-ec2" "${file}"
+
+    ## Setup spaces
+    juju reload-spaces
+    juju add-space isolated 172.31.254.0/24
+
+    ## Create machine 
+    ec2_add_multi_nic_machine
+    juju_machine_id=$(juju show-machine --format json | jq -r '.["machines"] | keys[0]')
+    
+    # Deploy test charm to dual-nic machine
+    juju deploy cs:~juju-qa/bionic/space-defender-1 --bind "defend-a=alpha defend-b=isolated" --to "${juju_machine_id}"
+    unit_index=$(get_unit_index "space-defender")
+    wait_for "space-defender" "$(idle_condition "space-defender" 0 "${unit_index}")"
+
+    assert_net_iface_for_endpoint_matches "space-defender" "defend-a" "ens5"
+    assert_net_iface_for_endpoint_matches "space-defender" "defend-b" "ens6"
+
+    assert_endpoint_binding_matches "space-defender" "" "alpha"
+    assert_endpoint_binding_matches "space-defender" "defend-a" "alpha"
+    assert_endpoint_binding_matches "space-defender" "defend-b" "isolated"
+
+    # Mutate bindings
+    juju bind space-defender defend-a=alpha defend-b=alpha
+    sleep 3
```

Comment:
Why the sleep, can we instead check the status for this? I think this will be flaky on some setups...

Reply:
I guess this is not needed here. I was probably thinking of the test with related charms where you need to wait for the uniter to pick up the config change, update the relation data and wait for it to be synced on the other end. I will remove it.